business/checkers/destinationrules/disabled_namespacewide_mtls_checker.go

```diff
@@ -21,11 +21,18 @@ func (m DisabledNamespaceWideMTLSChecker) Check() ([]*models.IstioCheck, bool) {
 
 	// otherwise, check among Policies for a rule enabling mTLS
 	for _, mp := range m.MTLSDetails.Policies {
-		// If Policy has mTLS enabled in the namespace
-		// traffic going through DestinationRule won't work
-		if strictMode := kubernetes.PolicyHasStrictMTLS(mp); strictMode {
-			check := models.Build("destinationrules.mtls.policymtlsenabled", "spec/trafficPolicy/tls/mode")
-			return append(validations, &check), false
+		if enabled, mode := kubernetes.PolicyHasMTLSEnabled(mp); enabled {
```

Comment:
I'm slightly confused with this change. Wouldn't that "PolicyHasStrictMTLS" be basically the same thing? Or are there other modes than "PERMISSIVE" && "STRICT" ?

Reply:
> Or are there other modes than "PERMISSIVE" && "STRICT" ?

No, there aren't more modes than that. 

> Wouldn't that "PolicyHasStrictMTLS" be basically the same thing?

No, because PolicyHasStrictMTLS returns only true for STRICT mode. And here we want to treat PERMISSIVE items too. If we found a Permissive policy, then the validation has to finish with no checks and true as validity.
If we don't take into account the Permissive mode, then validation checks for MeshPolicies and it might found a STRICT mode MeshPolicy and might return an invalid validation.


It fixes the Scenario 2:
MeshPolicy in STRICT mode + DestinationRule enabling mTLS mesh-wide (classic scenario)
Policy ns-level in PERMISSIVE mode + DR disabling mTLS ns-wide.
Problem: DR rule shows validation error: MeshPolicy in strict mode has found, should be permissive.
Expected: No validations shown for that DR.
Explanation: ns-level Policy in PERMISSIVE mode allows disabled mtls connections.

Perhaps another way to implement that would be (although I think is running almost same code twice):
```golang
// If there is a Policy in strict Mode then traffic from Destination Rule won't work
if strictMode := kubernetes.PolicyHasStrictMTLS(mp); strictMode {
	check := models.Build("destinationrules.mtls.policymtlsenabled", "spec/trafficPolicy/tls/mode")
	return append(validations, &check), false
}

// If we found a Policy in permissive mode, then traffic from Destination Rule will work
if permissiveMode := kubernetes.PolicyHasPermissiveMTLS(mp); permissiveMode {
	return validations, true
}
```

Reply:
Ah, the github view got me. It looked like the processing ended in both implementations.